var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -145,12 +147,10 @@ def install(self, spec, prefix):
             '-DTPL_ENABLE_MPI:BOOL=ON',
             '-DMPI_BASE_DIR:PATH=%s' % spec['mpi'].prefix,
             '-DTPL_ENABLE_BLAS=ON',
-            '-DBLAS_LIBRARY_NAMES=%s' % to_lib_name(
-                spec['blas'].blas_shared_lib),
+            '-DBLAS_LIBRARY_NAMES=%s' % ';'.join(blas.names),
             '-DBLAS_LIBRARY_DIRS=%s' % spec['blas'].prefix.lib,
             '-DTPL_ENABLE_LAPACK=ON',
-            '-DLAPACK_LIBRARY_NAMES=%s' % to_lib_name(
-                spec['lapack'].lapack_shared_lib),
+            '-DLAPACK_LIBRARY_NAMES=%s' % ';'.join(lapack.names),
             '-DLAPACK_LIBRARY_DIRS=%s' % spec['lapack'].prefix.lib,
```

Comment:
Should we be using dirnames for stuff like this? I don't think there are currently any BLAS/LAPACKs in Spack that keep their libraries in multiple directories, but if we have the functionality we might as well use it.